packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -132,7 +132,8 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
       styles: customStyles,
       upArrowButtonStyles: customUpArrowButtonStyles,
       downArrowButtonStyles: customDownArrowButtonStyles,
-      theme
+      theme,
+      onRenderIcon
```

Comment:
You're should pass the default function into the render function as the second param.

Also you can just set a default value to `this._defaultOnRenderIcon`, and call it without ternaries.

```tsx
let { onRenderIcon = this._defaultOnRenderIcon } = props;

{ onRenderIcon(this.props, this._defaultOnRenderIcon);
```

Reply:
Another thought, exposing this is just to allow us to support SVG icons in our app. This approach would require doing this in a bunch of places (like IContextualMenuItem and anywhere there is a chevron or ...). 

We could either 
a) Support SVG's in fabric natively by enhancing Icon/IconProps. This might not be straightforward because there are multiple ways people would want to consume SVGs (inline vs external) and it may actually require a pollyfill.

b) Expose a way for registering a custom renderer globally for all icon components. 

Thoughts @jair-cazarin?

Reply:
I really like the idea of extending registerIcons to support it, if we can make getIconClassName also work with svg. how are you doing it? Background image? can it be done through css?

Reply:
We are generating a sprite using svg Symbols and then using <use />tags  to reference individual symbols from the sprite. We are also using a "polyfill" called svg4everybody to inline them to that css can be applied.

Reply:
It's true that there are multiple ways to consume SVG's, I'll share a document where Jesse discuss some of them. We went with this approach mainly because we can cache the sprite and still be able to apply external css.

Reply:
If we enhance the Icon/IconProps, we could add a new svg iconType and leverage either the iconName or the src in imageProps to pass in the reference to the specific svg from the sprite. In the building of the Icon, if the type is svg we can use the <use> element to create everything correctly. (We can name the type something like svgUse if we want to make it more clear what it maps to)

Reply:
That sounds like a path worth exploring. I'll abandon the PR for now while we prototype.